bundles/org.openhab.binding.blink/src/main/java/org/openhab/binding/blink/internal/handler/AccountHandler.java

```diff
@@ -0,0 +1,402 @@
+            if (refreshToken)
+                setOffline(e);
+            return false;
+        }
+    }
+
+    /**
+     * Perform a homescreen update, either because refreshInterval has elapsed or due to an event.
+     *
+     * @param scheduled true if refreshInterval has elapsed, otherwise false
+     */
+    void refreshState(boolean scheduled) {
+        synchronized (refreshStateMonitor) {
+            try {
+                if (!scheduled && refreshStateJob != null) {
+                    refreshStateJob.cancel(true);
+                }
+                if (ensureBlinkAccount()) {
+                    loadDevices();
+                    loadEvents();
+                }
+            } finally {
+                refreshStateJob = scheduler.schedule(() -> refreshState(true), config.refreshInterval,
+                        TimeUnit.SECONDS);
+            }
+        }
+    }
+
+    public @Nullable BlinkHomescreen getDevices(boolean refresh) {
+        if (refresh || cachedHomescreen == null) {
+            refreshState(false);
+        }
+        return cachedHomescreen;
+    }
+
+    void loadDevices() {
+        logger.debug("Loading devices from Blink API");
+        try {
+            cachedHomescreen = blinkService.getDevices(blinkAccount);
+            fireHomescreenUpdate();
+        } catch (IOException e) {
+            setOffline(e);
+        }
+    }
+
+    void loadEvents() {
+        logger.debug("Loading events from Blink API");
+        try {
+            BlinkEvents events = blinkService.getEvents(blinkAccount, eventSince);
+            OffsetDateTime nextEventSince = eventSince;
+            for (BlinkEvents.Media mediaEvent : events.media) {
+                if (mediaEvent.deleted) {
+                    eventStore.remove(mediaEvent.id);
+                } else {
+                    eventStore.put(mediaEvent.id, mediaEvent);
+                }
+                if (nextEventSince.isBefore(mediaEvent.updated_at)) {
+                    nextEventSince = mediaEvent.updated_at;
+                }
+                // only trigger events if not polling for the first time
+                if (!eventSince.isEqual(EPOCH_UTC)) {
+                    fireMediaEvent(mediaEvent);
+                }
+            }
+            eventSince = nextEventSince;
+        } catch (IOException e) {
+            setOffline(e);
+        }
+    }
+
+    public final @Nullable BlinkAccount getBlinkAccount() {
+        return this.blinkAccount;
+    }
+
+    public final AccountConfiguration getConfiguration() {
+        return this.config;
+    }
+
+    public final String getGeneratedClientId() {
+        return generatedClientId;
+    }
+
+    BlinkCamera getCameraState(CameraConfiguration camera, boolean refresh) throws IOException {
+        Long cameraId = camera.cameraId;
+        BlinkHomescreen devices = getDevices(refresh);
+        if (blinkAccount == null) {
+            logger.error("Blink Account not set in bridge");
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be warn. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Please check all.